mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java

```diff
@@ -50,7 +54,8 @@ public ListenerType getType() {
     public void event(Event event) {
         if (event instanceof EventFactory.MailboxDeletionImpl) {
             try {
-                AnnotationMapper annotationMapper = mailboxSessionMapperFactory.getAnnotationMapper(event.getSession());
+                MailboxSession mailboxSession = mailboxManager.createSystemSession(event.getUser().asString());
```

Comment:
what about opening a session with a User instead of a String (in a further refactoring) ?

Reply:
or even better: not using a session at all ?

Reply:
How can I get an AnnotationMapper without using MailboxSession?

Reply:
it's a more general design comment. you're not supposed to change code for that right now.

Reply:
> opening a session with a User instead of a String (in a further refactoring)

:+1: